src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,12 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+
+        // If command changes addressbook data, save data to file
+        if (command.isMutating()) {
```

Comment:
This changes the behaviour of the `Logic#execute()` method. You need to update the header comment accordingly.